test/test_config_dump.sh

```diff
@@ -69,7 +69,7 @@ new "netconf xml"
 expectpart "$($clixon_netconf -q -f $cfg -C xml -o CLICON_MODULE_SET=42 -o CLICON_FEATURE="cmdline")" 0 '^<clixon-config xmlns="http://clicon.org/config">' "<CLICON_YANG_MAIN_DIR>$dir</CLICON_YANG_MAIN_DIR>" "<CLICON_MODULE_SET_ID>0</CLICON_MODULE_SET_ID>" "<autocli>" "<list-keyword-default>kw-nokey</list-keyword-default>" "<CLICON_FEATURE>orig</CLICON_FEATURE>" "<CLICON_FEATURE>cmdline</CLICON_FEATURE>" "<CLICON_FEATURE>extradir</CLICON_FEATURE>" --not-- "<CLICON_MODULE_SET_ID>42</CLICON_MODULE_SET_ID>"
 
 new "restconf xml"
-expectpart "$(sudo $clixon_restconf -f $cfg -C xml -o CLICON_MODULE_SET=42 -o CLICON_FEATURE="cmdline")" 0 '^<clixon-config xmlns="http://clicon.org/config">' "<CLICON_YANG_MAIN_DIR>$dir</CLICON_YANG_MAIN_DIR>" "<CLICON_MODULE_SET_ID>0</CLICON_MODULE_SET_ID>" "<autocli>" "<list-keyword-default>kw-nokey</list-keyword-default>" "<CLICON_FEATURE>orig</CLICON_FEATURE>" "<CLICON_FEATURE>cmdline</CLICON_FEATURE>" "<CLICON_FEATURE>extradir</CLICON_FEATURE>" --not-- "<CLICON_MODULE_SET_ID>42</CLICON_MODULE_SET_ID>"
+expectpart "$($clixon_restconf -f $cfg -C xml -o CLICON_MODULE_SET=42 -o CLICON_FEATURE="cmdline")" 0 '^<clixon-config xmlns="http://clicon.org/config">' "<CLICON_YANG_MAIN_DIR>$dir</CLICON_YANG_MAIN_DIR>" "<CLICON_MODULE_SET_ID>0</CLICON_MODULE_SET_ID>" "<autocli>" "<list-keyword-default>kw-nokey</list-keyword-default>" "<CLICON_FEATURE>orig</CLICON_FEATURE>" "<CLICON_FEATURE>cmdline</CLICON_FEATURE>" "<CLICON_FEATURE>extradir</CLICON_FEATURE>" --not-- "<CLICON_MODULE_SET_ID>42</CLICON_MODULE_SET_ID>"
```

Comment:
This change does not work with the mem.sh script.
```
olof@alarik> pattern=./test_config_dump.sh ./mem.sh restconf
Mem test restconf begin
=======================
valgrindfile:/tmp/tmp.TNVnO71p3j
Running ./test_config_dump.sh
Test 1(1) [cli xml]
Test 2(2) [backend xml]
Sep  5 09:23:24: clixon_backend: 2197031 Terminated retval:0
Test 3(3) [netconf xml]
Test 4(4) [restconf xml]
valgrind: Cannot create log file '/tmp/tmp.TNVnO71p3j': Permission denied

Error (1 != 0) in Test4 [restconf xml]:
```
Problem here is that the valgrind code assumes that restconf runs as root. One may possible change the valgrind code so that it may run as non-root (as valgrind clixon_cli does for example).

Reply:
I've pushed what I have.  I found some issues and rectified them.

If you want to pull the branch, add a tweak, and push it back I'm okay with that.

At this point, I'm seeing issues with `test_config_dump.sh` but it seems to be related with the output being generated mismatching what's expected.  I don't think it's related to my changes.

I'm not sure if `test_privileges.sh` is necessary, since after these changes all tests should be running as a non-privileged account and it becomes a moot point...  What we need to test that remains is running as a non-privileged user without dropping privileges, but that seems dangerous.

Reply:
Well, everything runs with `make test`, so we can call it good.  There are still some issues with running `./all.sh`, but those can be fixed separately.